sdk/servicebus/service-bus/test/utils/managementUtils.ts

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {
+  QueueOptions,
+  TopicOptions,
+  SubscriptionOptions,
+  ServiceBusAtomManagementClient,
+  delay
+} from "../../src";
+
+import { EnvVarKeys, getEnvVars } from "./envVarUtils";
+
+const env = getEnvVars();
+let client: ServiceBusAtomManagementClient;
+
+async function getManagementClient() {
+  if (client == undefined) {
+    client = await new ServiceBusAtomManagementClient(env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]);
+  }
+  return client;
+}
+
+/**
+ * Utility to apply retries to `create` management operations.
+ * Will ignore if given entity is already existing.
+ * @param createEntityOperation
+ * @param checkIfEntityExistsOperation
+ */
+async function retryCreateManagementEntityOperation(
+  createEntityOperation: any,
+  checkIfEntityExistsOperation: any
+) {
+  const retryAttempts = 5;
+  const retryDelayInMs = 1000;
+
+  let succeeded: boolean = false;
+  let count = 0;
+  while (count < retryAttempts) {
+    try {
+      const entityExists = await checkIfEntityExistsOperation();
+      if (entityExists) {
+        succeeded = true;
+        break;
+      }
+      await createEntityOperation();
+    } catch (err) {
+      // Ignore error and wait before retrying
+      await delay(retryDelayInMs);
+    } finally {
+      count++;
+    }
+  }
+
+  if (!succeeded) {
+    throw new Error("Error occurred while attempting to create a Service Bus entity");
+  }
+}
+
+/**
+ * Utility to apply retries to `delete` management operations.
+ * Will exit if given entity is already deleted.
+ * @param createEntityOperation
+ * @param checkIfEntityExistsOperation
+ */
+async function retryDeleteManagementEntityOperation(
+  deleteEntityOperation: any,
+  checkIfEntityExistsOperation: any
+) {
+  const retryAttempts = 5;
+  const retryDelayInMs = 1000;
+
+  let succeeded: boolean = false;
+  let count = 0;
+  while (count < retryAttempts) {
+    try {
+      const entityExists = await checkIfEntityExistsOperation();
+      if (!entityExists) {
+        succeeded = true;
+        break;
+      }
+      await deleteEntityOperation();
+    } catch (err) {
+      // Ignore error and wait before retrying
+      await delay(retryDelayInMs);
+    } finally {
+      count++;
+    }
+  }
+
+  if (!succeeded) {
+    throw new Error("Error occurred while attempting to delete a Service Bus entity");
+  }
```

Comment:
This logic is duplicated with the only thing different being the method we use to call (`deleteEntityOperation` vs `createEntityOperation`).

Consider separating this into a new method that takes as a parameter the function to call.

Reply:
Or take a callback that does the operation and it gets wrapped in the retry logic.

Reply:
The entity existing check is the inverse actually and logging message strings will be using a check too. I can still take a jab at refactoring

Reply:
Updated